nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ndarray/BaseSparseNDArray.java

```diff
@@ -326,14 +325,14 @@ public long lengthLong() {
         return length;
     }
 
-    protected void init(int[] shape) {
+    protected void init(long[] shape) {
 
         if (shape.length == 1) {
             rows = 1;
-            columns = shape[0];
+            columns = (int) shape[0];
```

Comment:
Let's also change those to `long` instead of casting.

Reply:
The reason I didn't use longs here, is that `INDArray.rows()`, `INDArray.columns()` and `INDArray.rank()` all return integers. So it seemed to be more in line to keep storing these as `ints`.

But we could also store them as longs and typecast in getter methods instead.

Reply:
Please change those to size(0) and size(1)! As for the rank I think that stays an int, @raver119?

Reply:
> Please change those to size(0) and size(1)

I'm not sure I complete understand what you are asking me to do?

Reply:
`INDArray.size(int dimension)` returns `long`, so let's use that instead of `rows()` and `columns()`!

Reply:
The method we are discussing does not use `INDArray.rows()`, rather the method uses a passed `long[] shape` argument to initialize. I know it is a bit weird, but I am just refactoring the existing code.

Reply:
Yes, that's fine, so we should also change the type of `rows` and `columns` to `long` at https://github.com/emillynge/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ndarray/BaseSparseNDArray.java#L49
among other places.

Reply:
Ok, I will do that.

Should we also change the underlying buffer to a LongBuffer While we're at it?
And should the constructor take both indices as `int[][]` **and** `long[][]`?

Reply:
Yup, that's the idea :) Thanks